Source/WebCore/page/Quirks.cpp

```diff
@@ -1574,4 +1574,24 @@ bool Quirks::shouldDisableDataURLPaddingValidation() const
     return *m_shouldDisableDataURLPaddingValidation;
 }
 
+bool Quirks::needsDisableDOMPasteAccessQuirk() const
+{
+    if (!needsQuirks())
+        return false;
+
+    if (m_needsDisableDOMPasteAccessQuirk)
+        return *m_needsDisableDOMPasteAccessQuirk;
+
+    m_needsDisableDOMPasteAccessQuirk = [&] {
+        if (!m_document)
+            return false;
+        auto* globalObject = m_document->globalObject();
+        if (!globalObject)
+            return false;
+        auto tableauPrepProperty = JSC::Identifier::fromString(globalObject->vm(), "tableauPrep"_s);
+        return globalObject->hasProperty(globalObject, tableauPrepProperty);
```

Comment:
This can execute arbitrary scripts and its script observable. Why not check the domain name first then check this property to limit the impact this quirk detection has?

Reply:
Good point —the intent was to make this quirk cover all Tableau instances, but I can limit it by host name (tableauanalytics.corp.apple.com) to avoid exposing any script-observable behaviors here.

Thanks for the review!

Reply:
Oh, Tableau can be deployed on third party domains? If that's the case I suppose the current quirk is okay.

Reply:
Indeed! Okay — I think I'll preserve the current behavior then (without limiting to domain as well). At least, this quirk check _only_ happens when we're about to dispatch DOM key events anyways, so it shouldn't introduce any weird new opportunities for arbitrary script execution.